Controllers/ListingController.cs

```diff
@@ -33,27 +33,25 @@ public ListingController(ILogger<ListingController> logger, IListingRepository l
         [Authorize]
         public ActionResult Post(Listing listing)
         {
-            var random = new Random();
-            listing.Id = random.Next(9999).ToString(); // temp fix for id generation, later this should be assigned in DB.
             listing.DatePublished = DateTime.UtcNow; //temp fix for curr date with form submit
 
-            string userId = User.FindFirst(ClaimTypes.Name)?.Value;
+            string userId = this.TryGetUserId();
             listing.OwnerId = userId;

```

Comment:
Should maybe do a null check for userId if authentication fails.

Reply:
Good point, I have added it. However, I believe the [Authorize] attribute should ensure that userId is not null, so maybe we should throw an exception in such a case as it is unexpected?

Reply:
Yeah, that's a good idea, didn't know that [Authorize] attribute checks whether id is null or not.

Reply:
Well, I'm not sure how the Authorize attribute works, but it checks that we have a request sent in with a token, and how I understand it, we get the userId based on that token.